From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Eric Reinsmidt <735251+ericreinsmidt@users.noreply.github.com>
Date: Sat, 10 Oct 2026 22:50:43 -0400
Subject: [PATCH] The kernel builds with GCC 10: dtc's lexer declared yylloc a
 second time, which GCC 10's -fno-common turns into a link error, so it is
 extern there, as mainline has had it since 2020

---
 scripts/dtc/dtc-lexer.l             | 2 +-
 scripts/dtc/dtc-lexer.lex.c_shipped | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/scripts/dtc/dtc-lexer.l b/scripts/dtc/dtc-lexer.l
index c60060304..bc40a01f7 100644
--- a/scripts/dtc/dtc-lexer.l
+++ b/scripts/dtc/dtc-lexer.l
@@ -38,7 +38,7 @@ LINECOMMENT	"//".*\n
 #include "srcpos.h"
 #include "dtc-parser.tab.h"
 
-YYLTYPE yylloc;
+extern YYLTYPE yylloc;
 extern bool treesource_error;
 
 /* CAUTION: this will stop working if we ever use yyless() or yyunput() */
diff --git a/scripts/dtc/dtc-lexer.lex.c_shipped b/scripts/dtc/dtc-lexer.lex.c_shipped
index cc7fa8248..3a9fb4817 100644
--- a/scripts/dtc/dtc-lexer.lex.c_shipped
+++ b/scripts/dtc/dtc-lexer.lex.c_shipped
@@ -631,7 +631,7 @@ char *yytext;
 #include "srcpos.h"
 #include "dtc-parser.tab.h"
 
-YYLTYPE yylloc;
+extern YYLTYPE yylloc;
 extern bool treesource_error;
 
 /* CAUTION: this will stop working if we ever use yyless() or yyunput() */
